lib/grammers-client/src/client/chats.rs

```diff
@@ -725,6 +725,55 @@ impl Client {
         let permissions = ParticipantPermissions::Channel(participant.participant);
         Ok(permissions)
     }
+    
+    /// join private chat
+    pub async fn accept_invite_link(
+        &mut self,
+        invite_link: &str,
+    ) -> Result<Option<Chat>, InvocationError> {
+        use tl::enums::Updates;
+        assert!(invite_link.starts_with("https://t.me/joinchat/"));
```

Comment:
We can't have asserts like this. The code should return an error if the link is not valid.

Reply:
To return an error, define a new error type, like auth::SignInError, is there any other way?

Reply:
I think we should create a fake RPC error like this:

https://github.com/Lonami/grammers/blob/c159fb086224342a4a45f719874cc675a0db5e46/lib/grammers-client/src/types/chats.rs#L69-L74

but with `name: "INVITE_HASH_INVALID"`.